test/e2e/util/kubevirt.go

```diff
@@ -0,0 +1,94 @@
+package util
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests"
+)
+
+// KubeVirtEnabled indicates if the test environment is
+// setup to allow usage of the KubeVirt platform.
+func KubeVirtEnabled() bool {
```

Comment:
What will set this env var? The test?

Maybe we can read the KV CRDs instead?

Reply:
We'll set it in the openshift ci config for the kubevirt test lane.

I want the test to be explicitly enabled or disabled via an env var in order to avoid false positive test results that simply silently skipped the kubevirt tests. This is based on experience. I've seen where we've attempted to create auto detection logic to determine when a test should run or not cause issues (unexpectedly skipped) which allowed regressions

Reply:
I don't think we shouldn't introduce an arbitrary pattern that rely on each single test checking an env variable.
Atm these are developed as regular go tests, the built in way to run specific tests is via go test -run "prefix" which should be ensure by the caller, e.g ci bin/test-e2e -test.v -test.timeout 0 -test.run='^TestAgent.*'.

Reply:
what you suggest breaks the aws run-all lane. We'd have to modify that test lane to opt in to specific tests, or we can go with the env var solution I'm proposing here which skips tests which require different environmental setup.

I'm fine with either direction. Which one makes the most sense for you all?

Reply:
I don't see an issue with dividing our test configs in CI right now and relying on `-test.run` to whitelist the suite for the given CI test definition. It's not clear we want all these tests in the blocking presubmits, for example. Maybe I'm misunderstanding the problem still.

Reply:
Another thinking I'm having now while looking at our code is all platforms should aim to pass this tests. We could keep tests names platform agnostic TestUpgradeControlPlane, TestAutoRepair...
Do the platform discrimination within a single func e.g hostedCluster := e2eutil.Create(Platform)Cluster based on whatever platorm the test is running on which can be passed through t parameter. Your e2e job will fail until is able to not, which makes sense. Thoughts?

Reply:
I'm with you, I don't think we can address it in the scope of this PR though. I started taking some steps in the general direction with https://github.com/openshift/hypershift/pull/884 (which unwinds most of the abstractions). Let's take the broader internal arch discussion elsewhere?

Reply:
ok @ironcladlou I see that's pretty much what you did here https://github.com/openshift/hypershift/pull/884/files

Reply:
Our periodics currently run all the tests in the e2e package and don't use `-test.run` to filter. So either that has to change (and the test names need updated so we can whitelist them), or we need another strategy to prevent this test from running there (which assumes AWS).

Reply:
My suggestion for now is to just move the env var test directly into TestKubeVirtCreateCluster. There's only one WIP test now, so I don't see the need to extract a whole utility file and codify it at this point.

Reply:
i moved the env var check to the kubevirt test and left a TODO note about the long term direction we're shooting for here.